src/components/ExceededCommentLength.js

```diff
@@ -14,47 +14,27 @@ const propTypes = {
     onExceededMaxCommentLength: PropTypes.func.isRequired,
 };
 
-class ExceededCommentLength extends PureComponent {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            commentLength: 0,
-        };
-
-        // By debouncing, we defer the calculation until there is a break in typing
-        this.updateCommentLength = debounce(this.updateCommentLength.bind(this), CONST.TIMING.COMMENT_LENGTH_DEBOUNCE_TIME);
-    }
-
-    componentDidMount() {
-        this.updateCommentLength();
+function ExceededCommentLength(props) {
+    const [commentLength, setCommentLength] = useState(0);
+    const updateCommentLength = useMemo(
+        () =>
+            debounce((comment, onExceededMaxCommentLength) => {
```

Comment:
In this case we are not returning the function straight away, but passing it to the `debounce` function.
Since the `useCallback` is not using a function as it's first argument, the `debounce` function will be evaluated on each re-render (so basically, each time the `text` props changes), but the result will always be ignored, since the dependency array is empty.

Is the `debounce` function too expensive to call on each re-render? I would argue that it is [a lot of code that would be run for each keystroke](https://github.com/lodash/lodash/blob/4.17.15/lodash.js#L10304-L10427), considering that we didn't do it before.

Reply:
I agree with @szebniok here, although I need to refresh my hooks knowledge too